modules/rostests/apitests/ntdll/RtlDebugInformation.c

```diff
@@ -0,0 +1,127 @@
+
+    Buffer = RtlCreateQueryDebugBuffer(1, FALSE);
+    ok(Buffer != NULL, "Unable to create default buffer\n");
+    if (!Buffer)
+        return;
+
+    ok_ptr(Buffer->ViewBaseClient, Buffer);
+    ok_hex(Buffer->Flags, 0);
+    ok(Buffer->OffsetFree == sizeof(*Buffer) || Buffer->OffsetFree == 0x60, "Expected %u or %u, got %lu\n", sizeof(*Buffer), 0x60, Buffer->OffsetFree);
+    ok_hex(Buffer->CommitSize, 0x1000);
+    ok_hex(Buffer->ViewSize, 0x1000);
+
+    Status = RtlDestroyQueryDebugBuffer(Buffer);
+    ok_hex(Status, STATUS_SUCCESS);
+
+    Buffer = RtlCreateQueryDebugBuffer(0x1000, FALSE);
+    ok(Buffer != NULL, "Unable to create default buffer\n");
+    if (!Buffer)
+        return;
+
+    ok_ptr(Buffer->ViewBaseClient, Buffer);
+    ok_hex(Buffer->Flags, 0);
+    ok(Buffer->OffsetFree == sizeof(*Buffer) || Buffer->OffsetFree == 0x60, "Expected %u or %u, got %lu\n", sizeof(*Buffer), 0x60, Buffer->OffsetFree);
+    ok_hex(Buffer->CommitSize, 0x1000);
+    ok_hex(Buffer->ViewSize, 0x1000);
+
+    Status = RtlDestroyQueryDebugBuffer(Buffer);
+    ok_hex(Status, STATUS_SUCCESS);
+
+    Buffer = RtlCreateQueryDebugBuffer(0x1001, FALSE);
+    ok(Buffer != NULL, "Unable to create default buffer\n");
+    if (!Buffer)
+        return;
+
+    ok_ptr(Buffer->ViewBaseClient, Buffer);
+    ok_hex(Buffer->Flags, 0);
+    ok(Buffer->OffsetFree == sizeof(*Buffer) || Buffer->OffsetFree == 0x60, "Expected %u or %u, got %lu\n", sizeof(*Buffer), 0x60, Buffer->OffsetFree);
+    ok_hex(Buffer->CommitSize, 0x1000);
+    ok_hex(Buffer->ViewSize, 0x2000);
+
+    Status = RtlDestroyQueryDebugBuffer(Buffer);
+    ok_hex(Status, STATUS_SUCCESS);
+
+    Buffer = RtlCreateQueryDebugBuffer(0x7fffffff, FALSE);
+    ok(Buffer == NULL, "Got a valid thing?\n");
+    if (Buffer)
+    {
+        Status = RtlDestroyQueryDebugBuffer(Buffer);
+        ok_hex(Status, STATUS_SUCCESS);
+    }
+}
+
+
+static void Test_ProcessModules(void)
+{
+    PRTL_DEBUG_INFORMATION Buffer;
+    NTSTATUS Status;
+    ULONG RequiredSize = 0;
+    PRTL_PROCESS_MODULES ExpectedModules;
+
+    Buffer = RtlCreateQueryDebugBuffer(0, FALSE);
+    ok(Buffer != NULL, "Unable to create default buffer\n");
+    if (!Buffer)
+        return;
+
+    Status = LdrQueryProcessModuleInformation(NULL, 0, &RequiredSize);
+    ok_hex(Status, STATUS_INFO_LENGTH_MISMATCH);
+
+    RequiredSize;
+    ExpectedModules = HeapAlloc(GetProcessHeap(), HEAP_ZERO_MEMORY, RequiredSize);
+
+    Status = LdrQueryProcessModuleInformation(ExpectedModules, RequiredSize, &RequiredSize);
+    ok_hex(Status, STATUS_SUCCESS);
+    if (NT_SUCCESS(Status))
+    {
+
+        Status = RtlQueryProcessDebugInformation(GetCurrentProcessId(), RTL_DEBUG_QUERY_MODULES, Buffer);
+        ok_hex(Status, STATUS_SUCCESS);
+        if (SUCCEEDED(Status))
+        {
+            ok(!memcmp(ExpectedModules, Buffer->Modules, RequiredSize), "Unexpected difference!\n");
+        }
+
+        HeapFree(GetProcessHeap(), 0, ExpectedModules);
```

Comment:
Memory leak if Status does not nt_succeed ?

Reply:
Indeed.